xgboost-spark document

CodingCat opened the pull request:
@yanboliang

CodingCat committed:
add back train method but mark as deprecated

CodingCat committed:
add back train method but mark as deprecated

CodingCat committed:
add back train method but mark as deprecated

CodingCat committed:
fix scalastyle error

CodingCat committed:
fix scalastyle error

CodingCat committed:
fix scalastyle error

CodingCat committed:
[jvm-packages] Maven central release  stuffs (#3401)

* add back train method but mark as deprecated

* add back train method but mark as deprecated

* fix scalastyle error

* fix scalastyle error

* maven central release

ngoyal2707 committed:
added code for instance based weighing for rank objectives (#3379)

* added code for instance based weighing for rank objectives

* Fix lint

a-johnston committed:
[jvm-packages] avoid use of Seq.apply in buildGroups (#3413)

RAMitchell committed:
Refactor python tests (#3410)

* Add unit test utility

* Refactor updater tests. Add coverage for histmaker.

chyunsu3 committed:
Fix wget for google tests in tests (#3414)

CI tests were failing because wget prompts "the user" for a response
whenever the google test archive is already on the disk.

Fix: Use `-nc` option to skip download when the archive already
exists

chyunsu3 committed:
Do not unzip google test archive if exists (#3416)

henrygouk committed:
Refactor of FastHistMaker to allow for custom regularisation methods (#3335)

* Refactor to allow for custom regularisation methods

* Implement compositional SplitEvaluator framework

* Fixed segfault when no monotone_constraints are supplied.

* Change pid to parentID

* test_monotone_constraints.py now passes

* Refactor ColMaker and DistColMaker to use SplitEvaluator

* Performance optimisation when no monotone_constraints specified

* Fix linter messages

* Fix a few more linter errors

* Update the amalgamation

* Add bounds check

* Add check for leaf node

* Fix linter error in param.h

* Fix clang-tidy errors on CI

* Fix incorrect function name

* Fix clang-tidy error in updater_fast_hist.cc

* Enable SSE2 for Win32 R MinGW

Addresses https://github.com/dmlc/xgboost/pull/3335#issuecomment-400535752

* Add contributor

pdesahb committed:
Fix tweedie handling of base_score (#3295)

* fix tweedie margin calculations

* add entry to contributors

cinqs committed:
Better doc for save_model() / load_model() (#3143)

Be clear that they do not save Python-specific attributes

Yunni committed:
Convert handle == nullptr from SegFault to user-friendly error. (#3021)

* Convert SegFault to user-friendly error.

* Apply the change to DMatrix API as well

RAMitchell committed:
Build universal wheels using GPU CI (#3424)

Commit:
Save and load model in sklearn API (#3192)

* Add (load|save)_model to XGBModel

* Add docstring

* Fix docstring

* Fix mixed use of space and tab

* Add a test

* Fix Flake8 style errors

owlas committed:
allow arbitrary cross validation fold indices (#3353)

* allow arbitrary cross validation fold indices

 - use training indices passed to `folds` parameter in `training.cv`
 - update doc string

* add tests for arbitrary fold indices

liuliang01 committed:
Add qid like ranklib format (#2749)

* add qid for https://github.com/dmlc/xgboost/issues/2748

* change names

* change spaces

* change qid to bst_uint type

* change qid type to size_t

* change qid first to SIZE_MAX

* change qid type from size_t to uint64_t

* update dmlc-core

* fix qids name error

* fix group_ptr_ error

* Style fix

* Add qid handling logic to SparsePage

* New MetaInfo format + backward compatibility fix

Old MetaInfo format (1.0) doesn't contain qid field. We still want to be able
to read from MetaInfo files saved in old format. Also, define a new format
(2.0) that contains the qid field. This way, we can distinguish files that
contain qid and those that do not.

* Update MetaInfo test

* Simply group assignment logic

* Explicitly set qid=nullptr in NativeDataIter

NativeDataIter's callback does not support qid field. Users of NativeDataIter
will need to call setGroup() function separately to set group information.

* Save qids_ in SaveBinary()

* Upgrade dmlc-core submodule

* Add a test for reading qid

* Add contributor

* Check the size of qids_

* Document qid format

StrikerRUS committed:
fixed MinGW missed dll (#3430)

yanboliang committed:
[jvm-packages] Expose nativeBooster for XGBoostClassificationModel and XGBoostRegressionModel. (#3428)

tqchen committed:
[DOCS] Update link to readme (#3437)

tqchen committed:
Update README.md

chyunsu3 committed:
Add a notice for binary PyPI wheel (#3443)

chyunsu3 committed:
Add callback interface to re-direct console output (#3438)

* Add callback interface to re-direct console output

* Exempt TrackerLogger from custom logging

* Fix lint

chyunsu3 committed:
Fix get_uint_info() (#3442)



* Add regression test

CodingCat committed:
[jvm-packages] disable booster setup for xgboost4j-spark (#3456)

* add back train method but mark as deprecated

* add back train method but mark as deprecated

* fix scalastyle error

* fix scalastyle error

* disable booster setup in spark

* check in parameter conversion

* fix compilation issue

* update exception type

ngoyal2707 committed:
[jvm-packages] removed old group_data from spark api (#3451)

kodonnell committed:
python sklearn api: defaulting to best_ntree_limit if defined, otherwise current behaviour (#3445)

* python sklearn api: defaulting to best_ntree_limit if defined, otherwise current behaviour

* Fix whitespace

chyunsu3 committed:
Document 0.72.1 version (#3458)

teju85 committed:
Upgrade cuda version to 9.2 for CI workflows (#3460)

- Needed by the issue #3404
 - as v9.1 doesn't have a nccl2 release

teju85 committed:
Upgrading to NCCL2 (#3404)

* Upgrading to NCCL2

* Part - II of NCCL2 upgradation

 - Doc updates to build with nccl2
 - Dockerfile.gpu update for a correct CI build with nccl2
 - Updated FindNccl package to have env-var NCCL_ROOT to take precedence

* Upgrading to v9.2 for CI workflow, since it has the nccl2 binaries available

* Added NCCL2 license + copy the nccl binaries into /usr location for the FindNccl module to find

* Set LD_LIBRARY_PATH variable to pick nccl2 binary at runtime

* Need the nccl2 library download instructions inside Dockerfile.release as well

* Use NCCL2 as a static library

yanboliang committed:
[JVM-packages] Support single instance prediction. (#3464)

* Support single instance prediction.

* Address comments.

henrygouk committed:
Add LASSO (#3429)

* Allow multiple split constraints

* Replace RidgePenalty with ElasticNet

* Add test for checking Ridge, LASSO, and Elastic Net are implemented

Commit:
upload doc

CodingCat commented:
sorry for the delay, I got bad cold in these days, 

let's first finish it here and then upstream to dmlc repo

RAMitchell committed:
Updates for GPU CI tests (#3467)

* Fail GPU CI after test failure

* Fix GPU linear tests

* Reduced number of GPU tests to speed up CI

* Remove static allocations of device memory

* Resolve illegal memory access for updater_fast_hist.cc

* Fix broken r tests dependency

* Update python install documentation for GPU

RAMitchell committed:
Resolve GPU bug on large files (#3472)

Remove calls to thrust copy, fix indexing bug

yanboliang commented on jvm-packages/xgboost4j-spark/docs/index.md:
```diff
@@ -0,0 +1,41 @@
+
+## Training 
+
+talk about different objective functions for different purposes
```
Explain how to set params with native XGBoost way and Spark MLlib way.

yanboliang commented on jvm-packages/xgboost4j-spark/docs/index.md:
```diff
@@ -0,0 +1,41 @@
+
+### Current Version of Gang Scheduling
+
+based on spark even listener
```
What about explain ```numWorkers``` as well? I found this param is a little confused and relevant to scheduling.

CodingCat replied:
I think we can talk about it in training part? as Gang Scheduling is an non-default behavior while numWorkers is something you must care about when using the system

yanboliang replied:
Either is ok, if we can address them more clear.

yanboliang reviewed (commented):
Looks very good overall, left two minor comments. Thanks.

CodingCat commented on jvm-packages/xgboost4j-spark/docs/index.md:
```diff
@@ -0,0 +1,41 @@
+
+## Data Preparation
+
+Data Frame based
```
@yanboliang I think we can choose a classic dataset here, maybe the ones here https://github.com/dmlc/xgboost/tree/master/demo/data

yanboliang replied:
Sound good.

CodingCat commented:
how you want to divide the workload? @yanboliang

KOLANICH committed:
Added .editorconfig (#3478)

yanboliang commented:
I'll take the section after ```Prediction```(included), you're responsible for the remaining part. What dow you think about it?

yanboliang committed:
Expose setCustomObj & setCustomEval for XGBoostClassifier & XGBoostRegressor. (#3486)

CodingCat commented:
sure,

chyunsu3 committed:
Doc modernization (#3474)

* Change doc build to reST exclusively

* Rewrite Intro doc in reST; create toctree

* Update parameter and contribute

* Convert tutorials to reST

* Convert Python tutorials to reST

* Convert CLI and Julia docs to reST

* Enable markdown for R vignettes

* Done migrating to reST

* Add guzzle_sphinx_theme to requirements

* Add breathe to requirements

* Fix search bar

* Add link to user forum

chyunsu3 committed:
Fix relpath in setup.py on Windows (#3493)

* Fix relpath in setup.py on Windows

Fixes #3480.

* Use only one lib file; use 4 space indent

KOLANICH committed:
Improved library loading a bit (#3481)

* Improved library loading a bit

* Fixed indentation.

* Fixes according to the discussion

* Moved the comment to a separate line.
* specified exception type

KOLANICH committed:
Added configuration for python into .editorconfig (#3494)

* Added configuration for python into .editorconfig

* Fixed forgotten change in the number of spaces

jqmp committed:
Add total_gain and total_cover importance measures (#3498)

Add `'total_gain'` and `'total_cover'` as possible `importance_type`
arguments to `Booster.get_score` in the Python package.

`get_score` already accepts a `'gain'` argument, which returns each
feature's average gain over all of its splits.  `'total_gain'` does the
same, but returns a total rather than an average.  This seems more
intuitively meaningful, and also matches the behavior of the R package's
`xgb.importance` function.

I also added an analogous `'total_cover'` command for consistency.

This should resolve #3484.

RAMitchell committed:
Correct mistake from dmatrix refactor (#3408)

CodingCat commented:
@yanboliang FYI, I have uploaded the latest version to github-based artifact repo

you can refer the dependency as https://github.com/dmlc/xgboost/tree/master/jvm-packages#access-snapshot-version

Commit:
docs
